tools/clang/unittests/HLSL/HlslTestUtils.h

```diff
@@ -235,6 +241,36 @@ inline bool ifdenorm_flushf_eq_or_nans(float a, float b) {
   return ifdenorm_flushf(a) == ifdenorm_flushf(b);
 }
 
+#else
+
+inline bool isdenorm(float f) {
+  return (std::numeric_limits<float>::denorm_min() <= f && f < std::numeric_limits<float>::min()) ||
+         (-std::numeric_limits<float>::min() <= f && f < -std::numeric_limits<float>::denorm_min());
```

Comment:
I think you need to swap the <= and < on the second line.

Reply:
Yep